data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java

```diff
@@ -450,4 +456,157 @@ public void getForkInfo_shouldHaveForkIfBlockRootNotEmpty() {
     assertThat(result.epoch).isEqualTo(beaconState.fork.epoch);
     assertThat(result.genesis_validators_root).isEqualTo(beaconState.genesis_validators_root);
   }
+
+  @Test
+  public void stateParameterToSlot_shouldParseHead() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(recentChainData, combinedChainDataClient);
+
+    Optional<UInt64> result = provider.stateParameterToSlot("head");
+    assertThat(result.isPresent()).isTrue();
+    assertThat(result).isEqualTo(recentChainData.getCurrentSlot());
+  }
+
+  @Test
+  public void stateParameterToSlot_shouldDetectInvalidValues() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(recentChainData, combinedChainDataClient);
+
+    assertThrows(NumberFormatException.class, () -> provider.stateParameterToSlot("hea"));
```

Comment:
nit: NumberFormatException seems a bit wrong for this error case?

Reply:
it's just the fallthrough - it looks for a bunch of pre-defined strings then tries a number if it's none of those. this was just a specific case where i typed 'hea' instead of 'head' and it failed and threw 500